jb/fftw/ut_time_delay_estimator_many.cpp

```diff
@@ -0,0 +1,335 @@
+    }
+    sum2[count] = acc_sum2;
+    expected_argmax[count] = static_cast<std::size_t>(delay);
+    expected_confidence[count] = static_cast<float>(nsamples);
+    count++;
+  }
+
+  // run TDE ....
+  (void)tested.estimate_delay(a, b, sum2, confidence, argmax);
+  // check argmax within delay +/- tol
+  BOOST_CHECK_MESSAGE(
+      jb::testing::tde_result_close_enough(argmax, expected_argmax, tol),
+      "argmax is not within tolerance("
+          << tol << ", argmax[0]=" << argmax[0]
+          << ", expected_argmax[0]=" << expected_argmax[0]);
+  // check confidence within nsamples +/- tol
+  BOOST_CHECK_MESSAGE(
+      jb::testing::tde_result_close_enough(
+          confidence, expected_confidence, 10 * tol),
+      "confidence is not within tolerance(" << 10 * tol << ")");
+}
+
+/**
+ * @test Verify that we can create and use a simple time delay estimator.
+ * dimension 1
+ */
+BOOST_AUTO_TEST_CASE(fftw_time_delay_estimator_many_1_dim_tde) {
+  int const nsamples = 1 << 15;
+  int const delay = 2500;
+  int const tol = 2;
+
+  using array_type = jb::fftw::aligned_multi_array<float, 1>;
+  using tested_type = jb::fftw::time_delay_estimator_many<array_type>;
+  using confidence_type = typename tested_type::confidence_type;
+  using argmax_type = typename tested_type::argmax_type;
+
+  array_type a(boost::extents[nsamples]);
+  array_type b(boost::extents[nsamples]);
+  confidence_type confidence(a);
+  // expected confidence result to compare within tolerance tol
+  confidence_type expected_confidence(a);
+
+  argmax_type argmax(a);
+  // expected argmax result to compare within tolerance tol
+  argmax_type expected_argmax(a);
+
+  using sum2_type = jb::fftw::tde_result<array_type, float>;
+  sum2_type sum2(b);
+
+  // construct the tested FTE
+  tested_type tested(a, b);
+
+  // fill the timeseries with triangle or square
+  jb::testing::create_triangle_timeseries(nsamples, b);
+
+  // a = delay shift timeseries b
+  float acc_sum2 = 0;
+  for (int k = 0; k != nsamples; ++k) {
+    a[k] = b[(k + delay) % nsamples];
+    acc_sum2 += a[k] * a[k];
+  }
+  sum2[0] = acc_sum2;
+  expected_argmax[0] = static_cast<std::size_t>(delay);
+  expected_confidence[0] = static_cast<float>(nsamples);
+
+  (void)tested.estimate_delay(a, b, sum2, confidence, argmax);
+  // check argmax within delay +/- tol
+  BOOST_CHECK_MESSAGE(
+      jb::testing::tde_result_close_enough(argmax, expected_argmax, tol),
+      "argmax is not within tolerance("
+          << tol << ", argmax[0]=" << argmax[0]
+          << ", expected_argmax[0]=" << expected_argmax[0]);
+  // check confidence within nsamples +/- tol
+  BOOST_CHECK_MESSAGE(
+      jb::testing::tde_result_close_enough(
+          confidence, expected_confidence, 10 * tol),
+      "confidence is not within tolerance(" << 10 * tol << ")");
+}
+
+/**
+ * @test Verify that FTE handles a timeseries filled with 0
+ * dimension 3
+ */
+BOOST_AUTO_TEST_CASE(fftw_time_delay_estimator_many_3_dim_tde_with_0) {
```

Comment:
ACK, done! I left  time_delay_estimator.hpp as it was, but _many now handles the functionality.